saleor/checkout/utils.py

```diff
@@ -738,6 +738,23 @@ def recalculate_cart_discount(cart, discounts, taxes):
         remove_voucher_from_cart(cart)
 
 
+def add_voucher_to_cart(voucher, cart):
+    """Add voucher data to cart.
+
+    Raise NotApplicable if voucher of given type cannot be applied."""
+    discount_amount = get_voucher_discount_for_cart(voucher, cart)
+    cart.voucher_code = voucher.code
+    cart.discount_name = voucher.name
+    cart.translated_discount_name = (
```

Comment:
You could simply use `cart.translated_discount_name = voucher.translated.name` - translations underneath default to original model field values.